src/layouts/EditHomepage/EditHomepage.jsx

```diff
@@ -373,23 +374,23 @@ const EditHomepage = ({ match }) => {
             textCardItemErrors.push(
               _.map(cards, () => getErrorValues(TEXTCARDS_ITEM_SECTION))
             )
-            if (textCardItemErrors.length === 0) {
-              textCardItemErrors.push([])
-            }
           }
 
+          // TODO: seems to load an existing state incorrectly into the errors object
           if (section.infocols) {
             sectionsErrors.push({
               infocols: getErrorValues(INFOCOLS_BLOCK_SECTION),
             })
             const { infoboxes } = section.infocols
             // Fill in dropdown elem errors array
+
             infocolInfoboxErrors.push(
               _.map(infoboxes, () => getErrorValues(INFOCOLS_INFOBOX_SECTION))
             )
-            if (infocolInfoboxErrors.length === 0) {
-              infocolInfoboxErrors.push([])
-            }
+          } else {
+            // populate for nested components
+            textCardItemErrors.push([])
+            infocolInfoboxErrors.push([])
```

Comment:
do we not need to do the checks here? (length === 0)

Reply:
the length checks were never there. I added it in the parent PR (IS-512) which was wrong. This logic exists to keep track of the errors with the indices representing the section